From ea7148731608d1253e164c98257dde2ccdf23ebd Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Tue, 4 Dec 2012 15:16:49 +0800
Subject: [PATCH 14/22] powerpc/sata: p2041si/p3041si: add missing compatible
 for sata workaround

A workaround is needed for sata controller version 1.0 on p2041/p3041/p5020,
but the compatible which the workaround based on is missing in new device
tree infrastructure, this will cause sata probe failed.

Signed-off-by: Shaohui Xie <Shaohui.Xie@freescale.com>
[Extracted from QorIQ-SDK-V1.3-SOURCE-20121114-yocto.iso]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/boot/dts/fsl/p2041si-post.dtsi |    6 ++++++
 arch/powerpc/boot/dts/fsl/p3041si-post.dtsi |    6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi b/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
index e1dcc1e..33bf082 100644
--- a/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p2041si-post.dtsi
@@ -330,7 +330,13 @@
 		};
 
 /include/ "qoriq-sata2-0.dtsi"
+	sata@220000 {
+		compatible = "fsl,p2041-sata", "fsl,pq-sata-v2";
+	};
 /include/ "qoriq-sata2-1.dtsi"
+	sata@221000 {
+		compatible = "fsl,p2041-sata", "fsl,pq-sata-v2";
+	};
 /include/ "qoriq-sec4.2-0.dtsi"
 /include/ "qoriq-pme-0.dtsi"
 /include/ "qoriq-rman-0.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi b/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
index 0761295..f6baf4a 100644
--- a/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/p3041si-post.dtsi
@@ -357,7 +357,13 @@
 		};
 
 /include/ "qoriq-sata2-0.dtsi"
+	sata@220000 {
+		compatible = "fsl,p3041-sata", "fsl,pq-sata-v2";
+	};
 /include/ "qoriq-sata2-1.dtsi"
+	sata@221000 {
+		compatible = "fsl,p3041-sata", "fsl,pq-sata-v2";
+	};
 /include/ "qoriq-sec4.2-0.dtsi"
 /include/ "qoriq-pme-0.dtsi"
 /include/ "qoriq-rman-0.dtsi"
-- 
1.7.9.7

